// @sedeh/into-pipes/esm2020/lib/slider/slider-pipe.module.mjs
import { NgModule, Inject, CUSTOM_ELEMENTS_SCHEMA } from '@angular/core';
import { CommonModule } from '@angular/common';
import { SliderComponent } from './slider.component';
import { ComponentPool } from '../common/component.pool';
import { CommonPipesModule } from '../common/common-pipes.module';
import * as i0 from "@angular/core";
import * as i1 from "../common/common-pipes.module";
import * as i2 from "../common/component.pool";
export class SliderIntoPipeModule {
    static forRoot() {
        return {
            ngModule: SliderIntoPipeModule,
            providers: []
        };
    }
    constructor(pool) {
        pool.registerComponent('slider', SliderComponent);
    }
}
SliderIntoPipeModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "15.2.10", ngImport: i0, type: SliderIntoPipeModule, deps: [{ token: ComponentPool }], target: i0.ɵɵFactoryTarget.NgModule });
SliderIntoPipeModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "15.2.10", ngImport: i0, type: SliderIntoPipeModule, declarations: [SliderComponent], imports: [CommonModule, i1.CommonPipesModule], exports: [SliderComponent] });
SliderIntoPipeModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "15.2.10", ngImport: i0, type: SliderIntoPipeModule, imports: [CommonModule, CommonPipesModule.forRoot()] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "15.2.10", ngImport: i0, type: SliderIntoPipeModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, CommonPipesModule.forRoot()],
                    declarations: [SliderComponent],
                    exports: [SliderComponent],
                    entryComponents: [SliderComponent],
                    schemas: [CUSTOM_ELEMENTS_SCHEMA]
                }]
        }], ctorParameters: function () { return [{ type: i2.ComponentPool, decorators: [{
                    type: Inject,
                    args: [ComponentPool]
                }] }]; } });
