// @sedeh/into-pipes/esm2020/lib/common/component.pool.mjs
import { Injectable } from '@angular/core';
import * as i0 from "@angular/core";
export class ComponentPool {
    constructor() {
        this.registeredPipes = {};
        this.registeredComponents = {};
        this.registeredServices = {};
    }
    registerPipeTransformation(name, pipe) {
        this.registeredPipes[name] = pipe;
    }
    registeredForPipeTransformationNamed(key) {
        return this.registeredPipes[key] !== undefined;
    }
    registeredPipeTransformation(key, content, args, callback, data) {
        const pipe = this.registeredPipes[key];
        return pipe ? pipe(content, args, callback, data) : null;
    }
    removePipeTransformation(key) {
        delete this.registeredPipes[key];
    }
    registerComponent(name, component) {
        this.registeredComponents[name] = component;
    }
    registeredForComponentWithNamed(name) {
        return this.registeredComponents[name] !== undefined;
    }
    registeredComponent(name, resolver, viewRefrence, el) {
        const component = name ? this.registeredComponents[name] : undefined;
        let result;
        if (component) {
            let componentFactory = resolver.resolveComponentFactory(component);
            const componentRef = viewRefrence.createComponent(componentFactory);
            const domElem = componentRef.hostView.rootNodes[0];
            el.appendChild(domElem);
            domElem.setAttribute("class", "into");
            result = componentRef.instance;
        }
        return result;
    }
    removeComponent(name) {
        delete this.registeredComponents[name];
    }
    registerServiceForComponent(name, service) {
        this.registeredServices[name] = service;
    }
    registeredServiceForComponent(name) {
        return this.registeredServices[name];
    }
    registeredForServiceNamed(name) {
        return this.registeredServices[name] !== undefined;
    }
    removeServiceForComponent(name) {
        delete this.registeredServices[name];
    }
}
ComponentPool.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "15.2.10", ngImport: i0, type: ComponentPool, deps: [], target: i0.ɵɵFactoryTarget.Injectable });
ComponentPool.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "15.2.10", ngImport: i0, type: ComponentPool });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "15.2.10", ngImport: i0, type: ComponentPool, decorators: [{
            type: Injectable
        }] });
