// @sedeh/into-pipes/esm2020/lib/common/into.pipe.mjs
import { Pipe } from '@angular/core';
import * as i0 from "@angular/core";
import * as i1 from "./component.pool";
export class InToPipe {
    constructor(pool) {
        this.pool = pool;
    }
    transform(content, list) {
        let result = content;
        list.split("|").map((item) => {
            result = this._transform(result, this.split(item));
        });
        return result;
    }
    split(item) {
        const matched = item.trim().match(/(?=\S)[^"\:]*(?:"[^\\"]*(?:\\[\:\S][^\\"]*)*"[^"\:]*)*/g);
        return matched ? matched.filter((x) => x.length) : [];
    }
    _transform(content, args) {
        let result = this.pool.registeredPipeTransformation(args[0], content, args, this._transform.bind(this));
        return result ? result : content;
    }
}
InToPipe.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "15.2.10", ngImport: i0, type: InToPipe, deps: [{ token: i1.ComponentPool }], target: i0.ɵɵFactoryTarget.Pipe });
InToPipe.ɵpipe = i0.ɵɵngDeclarePipe({ minVersion: "14.0.0", version: "15.2.10", ngImport: i0, type: InToPipe, name: "into" });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "15.2.10", ngImport: i0, type: InToPipe, decorators: [{
            type: Pipe,
            args: [{ name: 'into' }]
        }], ctorParameters: function () { return [{ type: i1.ComponentPool }]; } });
