// @sedeh/into-pipes/esm2020/lib/common/valueof.pipe.mjs
/*
* Defines a filter to convert a string into a map object.
*/
import { Pipe } from '@angular/core';
import * as i0 from "@angular/core";
export class ValueOfPipe {
    static transformationMethod() {
        const x = function (content, args, callback, data) {
            // valueof:key
            return new ValueOfPipe().transform(content, args.length > 1 ? args[1] : "");
        };
        return x;
    }
    valueOfSingle(source, key) {
        return source[key];
    }
    valueOfMultiple(sources, key) {
        const result = [];
        sources.map((source) => {
            result.push(this.valueOfSingle(source, key));
        });
        return result;
    }
    transform(object, ...args) {
        if ((typeof object === "string") || !(object instanceof Array)) {
            return this.valueOfSingle(object, args[0]);
        }
        return this.valueOfMultiple(object, args[0]);
    }
}
ValueOfPipe.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "15.2.10", ngImport: i0, type: ValueOfPipe, deps: [], target: i0.ɵɵFactoryTarget.Pipe });
ValueOfPipe.ɵpipe = i0.ɵɵngDeclarePipe({ minVersion: "14.0.0", version: "15.2.10", ngImport: i0, type: ValueOfPipe, name: "valueof" });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "15.2.10", ngImport: i0, type: ValueOfPipe, decorators: [{
            type: Pipe,
            args: [{ name: 'valueof' }]
        }] });
