// @sedeh/into-pipes/esm2020/lib/switch/switch.component.mjs
import { Component, ViewChild, Output, EventEmitter } from '@angular/core';
import * as i0 from "@angular/core";
import * as i1 from "@angular/common";
export class SwitchComponent {
    constructor() {
        this.onIntoComponentChange = new EventEmitter();
    }
    keyup(event) {
        const code = event.which;
        if (code === 13) {
            event.target.click();
        }
    }
    click(event) {
        event.stopPropagation();
        event.preventDefault();
        if (this.source === this.ideal) {
            this.source = this.original;
        }
        else {
            this.source = this.ideal;
        }
        this.onIntoComponentChange.emit({
            id: this.id,
            name: this.name,
            value: this.source,
            type: "switch",
            item: this.data
        });
        setTimeout(() => {
            if (this.check) {
                this.check.nativeElement.focus();
            }
            if (this.uncheck) {
                this.uncheck.nativeElement.focus();
            }
        }, 66);
    }
    transform(source, data, args) {
        this.ideal = args.length ? String(args[0]) : "";
        this.onText = args.length > 1 ? args[1] : 'ON';
        this.offText = args.length > 2 ? args[2] : 'OFF';
        this.source = String(source);
        this.data = data;
        this.original = this.source === this.ideal ? "" : source;
    }
}
SwitchComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "15.2.10", ngImport: i0, type: SwitchComponent, deps: [], target: i0.ɵɵFactoryTarget.Component });
SwitchComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "15.2.10", type: SwitchComponent, selector: "switch-component", outputs: { onIntoComponentChange: "onIntoComponentChange" }, viewQueries: [{ propertyName: "check", first: true, predicate: ["check"], descendants: true }, { propertyName: "uncheck", first: true, predicate: ["uncheck"], descendants: true }], ngImport: i0, template: `
      <span *ngIf="source === ideal" 
          #check tabindex="0" 
          class="switch fa fa-toggle-on" 
          aria-checked="true"
          role="checkbox"
          (keyup)="keyup($event)" 
          (click)="click($event)"></span>
      <span *ngIf="source !== ideal"
          #uncheck tabindex="0" 
          class="switch fa fa-toggle-off" 
          aria-checked="false"
          role="checkbox"
          (keyup)="keyup($event)" 
          (click)="click($event)"></span>
      <span class="text" [class.selected]="source === ideal" 
        [textContent]="source === ideal ? onText : offText"></span>

    `, isInline: true, styles: [":host{display:table;float:left;min-height:23px}:host .switch{font-size:1.4rem;cursor:pointer;float:left}:host .switch:hover{color:#fabdab}:host .switch.fa-toggle-on{color:green}:host .switch.fa-toggle-off{color:red}:host .text{font-size:1.2rem;text-transform:uppercase;float:left;margin-left:5px;color:red}:host .text.selected{color:green}\n"], dependencies: [{ kind: "directive", type: i1.NgIf, selector: "[ngIf]", inputs: ["ngIf", "ngIfThen", "ngIfElse"] }] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "15.2.10", ngImport: i0, type: SwitchComponent, decorators: [{
            type: Component,
            args: [{ selector: 'switch-component', template: `
      <span *ngIf="source === ideal" 
          #check tabindex="0" 
          class="switch fa fa-toggle-on" 
          aria-checked="true"
          role="checkbox"
          (keyup)="keyup($event)" 
          (click)="click($event)"></span>
      <span *ngIf="source !== ideal"
          #uncheck tabindex="0" 
          class="switch fa fa-toggle-off" 
          aria-checked="false"
          role="checkbox"
          (keyup)="keyup($event)" 
          (click)="click($event)"></span>
      <span class="text" [class.selected]="source === ideal" 
        [textContent]="source === ideal ? onText : offText"></span>

    `, styles: [":host{display:table;float:left;min-height:23px}:host .switch{font-size:1.4rem;cursor:pointer;float:left}:host .switch:hover{color:#fabdab}:host .switch.fa-toggle-on{color:green}:host .switch.fa-toggle-off{color:red}:host .text{font-size:1.2rem;text-transform:uppercase;float:left;margin-left:5px;color:red}:host .text.selected{color:green}\n"] }]
        }], propDecorators: { check: [{
                type: ViewChild,
                args: ["check"]
            }], uncheck: [{
                type: ViewChild,
                args: ["uncheck"]
            }], onIntoComponentChange: [{
                type: Output,
                args: ["onIntoComponentChange"]
            }] } });
