// @sedeh/into-pipes/esm2020/lib/pipe.module.mjs
import { NgModule, CUSTOM_ELEMENTS_SCHEMA } from '@angular/core';
import { CommonModule } from '@angular/common';
import { CommonPipesModule } from './common/common-pipes.module';
import { AddressIntoPipeModule } from './address/addess-pipe.module';
import { AudioIntoPipeModule } from './audio/audio-pipe.module';
import { CalendarIntoPipeModule } from './calendar/calendar-pipe.module';
import { CheckboxIntoPipeModule } from './checkbox/checkbox-pipe.module';
import { EmailIntoPipeModule } from './email/email-pipe.module';
import { FontIntoPipeModule } from './font/font-pipe.module';
import { ImageIntoPipeModule } from './image/image-pipe.module';
import { InputIntoPipeModule } from './input/input-pipe.module';
import { InputGroupIntoPipeModule } from './inputgroup/inputgroup-pipe.module';
import { JsonIntoPipeModule } from './json/json-pipe.module';
import { LastUpdateIntoPipeModule } from './lastupdate/lastupdate-pipe.module';
import { LikeIntoPipeModule } from './like/like-pipe.module';
import { LinkIntoPipeModule } from './link/link-pipe.module';
import { PhoneIntoPipeModule } from './phone/phone-pipe.module';
import { RatingIntoPipeModule } from './rating/rating-pipe.module';
import { NoticeIntoPipeModule } from './notice/notice-pipe.module';
import { SelectIntoPipeModule } from './select/select-pipe.module';
import { ShareIntoPipeModule } from './share/share-pipe.module';
import { SpanIntoPipeModule } from './span/span-pipe.module';
import { SwitchIntoPipeModule } from './switch/switch-pipe.module';
import { TableIntoPipeModule } from './table/table-pipe.module';
import { TextIntoPipeModule } from './text/text-pipe.module';
import { VideoIntoPipeModule } from './video/video-pipe.module';
import { SliderIntoPipeModule } from './slider/slider-pipe.module';
import * as i0 from "@angular/core";
import * as i1 from "./common/common-pipes.module";
import * as i2 from "./address/addess-pipe.module";
import * as i3 from "./audio/audio-pipe.module";
import * as i4 from "./calendar/calendar-pipe.module";
import * as i5 from "./checkbox/checkbox-pipe.module";
import * as i6 from "./email/email-pipe.module";
import * as i7 from "./font/font-pipe.module";
import * as i8 from "./image/image-pipe.module";
import * as i9 from "./input/input-pipe.module";
import * as i10 from "./inputgroup/inputgroup-pipe.module";
import * as i11 from "./json/json-pipe.module";
import * as i12 from "./lastupdate/lastupdate-pipe.module";
import * as i13 from "./like/like-pipe.module";
import * as i14 from "./link/link-pipe.module";
import * as i15 from "./phone/phone-pipe.module";
import * as i16 from "./rating/rating-pipe.module";
import * as i17 from "./notice/notice-pipe.module";
import * as i18 from "./select/select-pipe.module";
import * as i19 from "./share/share-pipe.module";
import * as i20 from "./slider/slider-pipe.module";
import * as i21 from "./span/span-pipe.module";
import * as i22 from "./switch/switch-pipe.module";
import * as i23 from "./table/table-pipe.module";
import * as i24 from "./text/text-pipe.module";
import * as i25 from "./video/video-pipe.module";
export class IntoPipeModule {
}
IntoPipeModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "15.2.10", ngImport: i0, type: IntoPipeModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
IntoPipeModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "15.2.10", ngImport: i0, type: IntoPipeModule, imports: [CommonModule, i1.CommonPipesModule, i2.AddressIntoPipeModule, i3.AudioIntoPipeModule, i4.CalendarIntoPipeModule, i5.CheckboxIntoPipeModule, i6.EmailIntoPipeModule, i7.FontIntoPipeModule, i8.ImageIntoPipeModule, i9.InputIntoPipeModule, i10.InputGroupIntoPipeModule, i11.JsonIntoPipeModule, i12.LastUpdateIntoPipeModule, i13.LikeIntoPipeModule, i14.LinkIntoPipeModule, i15.PhoneIntoPipeModule, i16.RatingIntoPipeModule, i17.NoticeIntoPipeModule, i18.SelectIntoPipeModule, i19.ShareIntoPipeModule, i20.SliderIntoPipeModule, i21.SpanIntoPipeModule, i22.SwitchIntoPipeModule, i23.TableIntoPipeModule, i24.TextIntoPipeModule, i25.VideoIntoPipeModule], exports: [CommonPipesModule,
        AddressIntoPipeModule,
        AudioIntoPipeModule,
        CalendarIntoPipeModule,
        CheckboxIntoPipeModule,
        EmailIntoPipeModule,
        FontIntoPipeModule,
        ImageIntoPipeModule,
        InputIntoPipeModule,
        InputGroupIntoPipeModule,
        JsonIntoPipeModule,
        LastUpdateIntoPipeModule,
        LikeIntoPipeModule,
        LinkIntoPipeModule,
        PhoneIntoPipeModule,
        RatingIntoPipeModule,
        NoticeIntoPipeModule,
        SelectIntoPipeModule,
        ShareIntoPipeModule,
        SliderIntoPipeModule,
        SpanIntoPipeModule,
        SwitchIntoPipeModule,
        TableIntoPipeModule,
        TableIntoPipeModule,
        TextIntoPipeModule,
        VideoIntoPipeModule] });
IntoPipeModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "15.2.10", ngImport: i0, type: IntoPipeModule, imports: [CommonModule,
        CommonPipesModule.forRoot(),
        AddressIntoPipeModule.forRoot(),
        AudioIntoPipeModule.forRoot(),
        CalendarIntoPipeModule.forRoot(),
        CheckboxIntoPipeModule.forRoot(),
        EmailIntoPipeModule.forRoot(),
        FontIntoPipeModule.forRoot(),
        ImageIntoPipeModule.forRoot(),
        InputIntoPipeModule.forRoot(),
        InputGroupIntoPipeModule.forRoot(),
        JsonIntoPipeModule.forRoot(),
        LastUpdateIntoPipeModule.forRoot(),
        LikeIntoPipeModule.forRoot(),
        LinkIntoPipeModule.forRoot(),
        PhoneIntoPipeModule.forRoot(),
        RatingIntoPipeModule.forRoot(),
        NoticeIntoPipeModule.forRoot(),
        SelectIntoPipeModule.forRoot(),
        ShareIntoPipeModule.forRoot(),
        SliderIntoPipeModule.forRoot(),
        SpanIntoPipeModule.forRoot(),
        SwitchIntoPipeModule.forRoot(),
        TableIntoPipeModule.forRoot(),
        TextIntoPipeModule.forRoot(),
        VideoIntoPipeModule.forRoot(), CommonPipesModule,
        AddressIntoPipeModule,
        AudioIntoPipeModule,
        CalendarIntoPipeModule,
        CheckboxIntoPipeModule,
        EmailIntoPipeModule,
        FontIntoPipeModule,
        ImageIntoPipeModule,
        InputIntoPipeModule,
        InputGroupIntoPipeModule,
        JsonIntoPipeModule,
        LastUpdateIntoPipeModule,
        LikeIntoPipeModule,
        LinkIntoPipeModule,
        PhoneIntoPipeModule,
        RatingIntoPipeModule,
        NoticeIntoPipeModule,
        SelectIntoPipeModule,
        ShareIntoPipeModule,
        SliderIntoPipeModule,
        SpanIntoPipeModule,
        SwitchIntoPipeModule,
        TableIntoPipeModule,
        TableIntoPipeModule,
        TextIntoPipeModule,
        VideoIntoPipeModule] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "15.2.10", ngImport: i0, type: IntoPipeModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [
                        CommonModule,
                        CommonPipesModule.forRoot(),
                        AddressIntoPipeModule.forRoot(),
                        AudioIntoPipeModule.forRoot(),
                        CalendarIntoPipeModule.forRoot(),
                        CheckboxIntoPipeModule.forRoot(),
                        EmailIntoPipeModule.forRoot(),
                        FontIntoPipeModule.forRoot(),
                        ImageIntoPipeModule.forRoot(),
                        InputIntoPipeModule.forRoot(),
                        InputGroupIntoPipeModule.forRoot(),
                        JsonIntoPipeModule.forRoot(),
                        LastUpdateIntoPipeModule.forRoot(),
                        LikeIntoPipeModule.forRoot(),
                        LinkIntoPipeModule.forRoot(),
                        PhoneIntoPipeModule.forRoot(),
                        RatingIntoPipeModule.forRoot(),
                        NoticeIntoPipeModule.forRoot(),
                        SelectIntoPipeModule.forRoot(),
                        ShareIntoPipeModule.forRoot(),
                        SliderIntoPipeModule.forRoot(),
                        SpanIntoPipeModule.forRoot(),
                        SwitchIntoPipeModule.forRoot(),
                        TableIntoPipeModule.forRoot(),
                        TextIntoPipeModule.forRoot(),
                        VideoIntoPipeModule.forRoot()
                    ],
                    declarations: [],
                    exports: [
                        CommonPipesModule,
                        AddressIntoPipeModule,
                        AudioIntoPipeModule,
                        CalendarIntoPipeModule,
                        CheckboxIntoPipeModule,
                        EmailIntoPipeModule,
                        FontIntoPipeModule,
                        ImageIntoPipeModule,
                        InputIntoPipeModule,
                        InputGroupIntoPipeModule,
                        JsonIntoPipeModule,
                        LastUpdateIntoPipeModule,
                        LikeIntoPipeModule,
                        LinkIntoPipeModule,
                        PhoneIntoPipeModule,
                        RatingIntoPipeModule,
                        NoticeIntoPipeModule,
                        SelectIntoPipeModule,
                        ShareIntoPipeModule,
                        SliderIntoPipeModule,
                        SpanIntoPipeModule,
                        SwitchIntoPipeModule,
                        TableIntoPipeModule,
                        TableIntoPipeModule,
                        TextIntoPipeModule,
                        VideoIntoPipeModule
                    ],
                    entryComponents: [],
                    providers: [],
                    schemas: [CUSTOM_ELEMENTS_SCHEMA]
                }]
        }] });
