// @sedeh/wizard-storage/esm2020/lib/wizard-storage.module.mjs
import { NgModule, CUSTOM_ELEMENTS_SCHEMA } from '@angular/core';
import { CommonModule } from '@angular/common';
import { WizardStorageService } from './wizard-storage.service';
import { WizardStorageDirective } from './wizard-storage.directive';
import * as i0 from "@angular/core";
export class WizardStorageModule {
    static forRoot() {
        return {
            ngModule: WizardStorageModule,
            providers: [
                WizardStorageService
            ]
        };
    }
}
WizardStorageModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "15.2.10", ngImport: i0, type: WizardStorageModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
WizardStorageModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "15.2.10", ngImport: i0, type: WizardStorageModule, declarations: [WizardStorageDirective], imports: [CommonModule], exports: [WizardStorageDirective] });
WizardStorageModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "15.2.10", ngImport: i0, type: WizardStorageModule, providers: [
        WizardStorageService
    ], imports: [CommonModule] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "15.2.10", ngImport: i0, type: WizardStorageModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [
                        WizardStorageDirective
                    ],
                    exports: [
                        WizardStorageDirective
                    ],
                    imports: [
                        CommonModule
                    ],
                    providers: [
                        WizardStorageService
                    ],
                    schemas: [CUSTOM_ELEMENTS_SCHEMA]
                }]
        }] });
