// @sedeh/wizard-storage/esm2020/lib/wizard-storage.directive.mjs
import { Directive, Output, HostListener, EventEmitter } from '@angular/core';
import * as i0 from "@angular/core";
import * as i1 from "./wizard-storage.service";
export class WizardStorageDirective {
    // Will listen to localStorage changes made
    // by other applications.
    onHover(event) {
        this.wizardStorage.emit({
            key: event.key,
            oldValue: this.wizardService.toJson(event.oldValue),
            newValue: this.wizardService.toJson(event.newValue),
            url: event.url
        });
    }
    constructor(wizardService) {
        this.wizardService = wizardService;
        this.wizardStorage = new EventEmitter();
    }
}
WizardStorageDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "15.2.10", ngImport: i0, type: WizardStorageDirective, deps: [{ token: i1.WizardStorageService }], target: i0.ɵɵFactoryTarget.Directive });
WizardStorageDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "14.0.0", version: "15.2.10", type: WizardStorageDirective, selector: "[wizardStorage]", outputs: { wizardStorage: "wizardStorage" }, host: { listeners: { "window:storage": "onHover($event)" } }, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "15.2.10", ngImport: i0, type: WizardStorageDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: '[wizardStorage]'
                }]
        }], ctorParameters: function () { return [{ type: i1.WizardStorageService }]; }, propDecorators: { onHover: [{
                type: HostListener,
                args: ['window:storage', ['$event']]
            }], wizardStorage: [{
                type: Output
            }] } });
