// @sedeh/into-pipes/esm2020/lib/notice/notice.component.mjs
import { Component, EventEmitter, HostListener } from '@angular/core';
import * as i0 from "@angular/core";
export class NoticeComponent {
    constructor(el) {
        this.el = el;
        this.onIntoComponentChange = new EventEmitter();
        el.nativeElement.setAttribute('tabindex', '0');
    }
    keyup(event) {
        const code = event.which;
        event.stopPropagation();
        event.preventDefault();
        if (code === 13) {
            event.target.click();
        }
    }
    click() {
        this.onIntoComponentChange.emit({
            id: this.id,
            name: this.name,
            value: this.source,
            type: 'click',
            item: 'notice'
        });
    }
    transform(source, data, args) {
        this.message = args.length ? args[0] : undefined;
        this.source = source;
        this.el.nativeElement.setAttribute('title', this.message);
    }
}
NoticeComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "15.2.10", ngImport: i0, type: NoticeComponent, deps: [{ token: i0.ElementRef }], target: i0.ɵɵFactoryTarget.Component });
NoticeComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "15.2.10", type: NoticeComponent, selector: "notice-component", host: { listeners: { "keyup": "keyup($event)", "click": "click()" } }, ngImport: i0, template: `
        <span class='fa fa-bell' aria-hidden='true'></span>
        <span class='notice' [textContent]="source"></span>
    `, isInline: true, styles: [":host{display:table;position:relative;float:left;cursor:pointer}:host .fa{font-size:1rem}:host:hover{color:red}:host:hover .fa{color:red}:host:hover .notice{background-color:#000}.notice{display:table;width:17px;height:15px;border-radius:50%;text-align:center;background-color:#379bffe6;float:right;font-size:.7rem;line-height:1rem;margin-left:-5px;z-index:2;color:#fff}\n"] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "15.2.10", ngImport: i0, type: NoticeComponent, decorators: [{
            type: Component,
            args: [{ selector: 'notice-component', template: `
        <span class='fa fa-bell' aria-hidden='true'></span>
        <span class='notice' [textContent]="source"></span>
    `, styles: [":host{display:table;position:relative;float:left;cursor:pointer}:host .fa{font-size:1rem}:host:hover{color:red}:host:hover .fa{color:red}:host:hover .notice{background-color:#000}.notice{display:table;width:17px;height:15px;border-radius:50%;text-align:center;background-color:#379bffe6;float:right;font-size:.7rem;line-height:1rem;margin-left:-5px;z-index:2;color:#fff}\n"] }]
        }], ctorParameters: function () { return [{ type: i0.ElementRef }]; }, propDecorators: { keyup: [{
                type: HostListener,
                args: ['keyup', ['$event']]
            }], click: [{
                type: HostListener,
                args: ['click', []]
            }] } });
