// @sedeh/into-pipes/esm2020/lib/notice/notice.pipe.mjs
/*
* Defines a filter to convert url into a raiting display.
*/
import { Pipe } from '@angular/core';
import * as i0 from "@angular/core";
export class NoticePipe {
    static transformationMethod() {
        const x = function (content, args, callback, data) {
            return new NoticePipe().transform(content, "");
        };
        return x;
    }
    noticeString(source, message) {
        return `
        <span style='display:table;possition:relative;float:left' alt='` + message + `'>
          <span class='fa fa-star' aria-hidden='true'></span>
          <span style='display: table;width: 17px;height: 15px;border-radius: 50%;text-align: center;background-color: rgba(200,200,200,0.2);float: right;font-size: 0.8rem;margin-left: -5px'>` +
            source +
            ` </span>
        </span>`;
    }
    transform(source, ...args) {
        const message = args.length ? args[0] : undefined;
        if ((typeof source === "string") || !(source instanceof Array)) {
            return this.noticeString(source, message);
        }
        else {
            const result = [];
            source.map((item) => {
                result.push(this.noticeString(item, message));
            });
            return result;
        }
    }
}
NoticePipe.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "15.2.10", ngImport: i0, type: NoticePipe, deps: [], target: i0.ɵɵFactoryTarget.Pipe });
NoticePipe.ɵpipe = i0.ɵɵngDeclarePipe({ minVersion: "14.0.0", version: "15.2.10", ngImport: i0, type: NoticePipe, name: "notice" });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "15.2.10", ngImport: i0, type: NoticePipe, decorators: [{
            type: Pipe,
            args: [{ name: 'notice' }]
        }] });
