// @sedeh/into-pipes/esm2020/lib/rating/rating.pipe.mjs
/*
* Defines a filter to convert url into a raiting display.
*/
import { Pipe } from '@angular/core';
import * as i0 from "@angular/core";
export class RatingPipe {
    static transformationMethod() {
        const x = function (content, args, callback, data) {
            return new RatingPipe().transform(content, "");
        };
        return x;
    }
    rateString(source, multiStart) {
        const value = parseInt(source, 10);
        const float = parseFloat(source);
        let x = "<span class='rating'>";
        if (multiStart) {
            for (let i = 0; i < value; i++) {
                x += "<span class='fa fa-star' aria-hidden='true'></span>";
            }
            if (float !== value) {
                x += "<span class='fa fa-star-half' aria-hidden='true'></span>";
            }
        }
        else {
            x += "<span class='fa fa-star' aria-hidden='true'></span>";
        }
        x += "</span><span class='rate-value'>" + source + "</span>";
        return x;
    }
    transform(source, ...args) {
        const singleStar = args.length ? (args[0] === 'true') : false;
        if ((typeof source === "string") || !(source instanceof Array)) {
            return this.rateString(source, singleStar);
        }
        else {
            const result = [];
            source.map((item) => {
                result.push(this.rateString(item, singleStar));
            });
            return result;
        }
    }
}
RatingPipe.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "15.2.10", ngImport: i0, type: RatingPipe, deps: [], target: i0.ɵɵFactoryTarget.Pipe });
RatingPipe.ɵpipe = i0.ɵɵngDeclarePipe({ minVersion: "14.0.0", version: "15.2.10", ngImport: i0, type: RatingPipe, name: "raiting" });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "15.2.10", ngImport: i0, type: RatingPipe, decorators: [{
            type: Pipe,
            args: [{ name: 'raiting' }]
        }] });
