// @sedeh/into-pipes/esm2020/lib/input/input-pipe.module.mjs
import { NgModule, Inject, CUSTOM_ELEMENTS_SCHEMA } from '@angular/core';
import { CommonModule } from '@angular/common';
import { InputComponent } from './input.component';
import { ComponentPool } from '../common/component.pool';
import { CommonPipesModule } from '../common/common-pipes.module';
import * as i0 from "@angular/core";
import * as i1 from "../common/common-pipes.module";
import * as i2 from "../common/component.pool";
export class InputIntoPipeModule {
    static forRoot() {
        return {
            ngModule: InputIntoPipeModule,
            providers: []
        };
    }
    constructor(pool) {
        pool.registerComponent('input', InputComponent);
    }
}
InputIntoPipeModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "15.2.10", ngImport: i0, type: InputIntoPipeModule, deps: [{ token: ComponentPool }], target: i0.ɵɵFactoryTarget.NgModule });
InputIntoPipeModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "15.2.10", ngImport: i0, type: InputIntoPipeModule, declarations: [InputComponent], imports: [CommonModule, i1.CommonPipesModule], exports: [InputComponent] });
InputIntoPipeModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "15.2.10", ngImport: i0, type: InputIntoPipeModule, imports: [CommonModule, CommonPipesModule.forRoot()] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "15.2.10", ngImport: i0, type: InputIntoPipeModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, CommonPipesModule.forRoot()],
                    declarations: [InputComponent],
                    exports: [InputComponent],
                    entryComponents: [InputComponent],
                    schemas: [CUSTOM_ELEMENTS_SCHEMA]
                }]
        }], ctorParameters: function () { return [{ type: i2.ComponentPool, decorators: [{
                    type: Inject,
                    args: [ComponentPool]
                }] }]; } });
